### PR TITLE
Remove init-containers in LVM mode

### LINKED ISSUE #532: merge pmem-ns-init and pmem-vgm into pmem-csi-driver
The information shown in those setup stages (these run in LVM mode only) can be helpful for debugging issues and monitoring storage devices state.

### PR DESCRIPTION
**Pull request by avalluri**
As we are using 'shared/Bidirectional' mounts for LVM driver also, we
could create the needed physical devices(namespaces, volume groups) in
the same driver session. This invalidates the usage of init-containers.

Few advantages of merging the init-containers functionality to driver code:
 - smaller container image size
 - allows us to implement features that are common to both modes(ex:
 pmemPercentage)

FIXES #532

### CODE REVIEW

File: docs/install.md [hunk 1]
```diff
@@ -341,10 +341,10 @@ for `kubectl kustomize`. For example:
          path: lvm-parameters-patch.yaml
      EOF
      $ cat >my-pmem-csi-deployment/lvm-parameters-patch.yaml <<EOF
-     # pmem-ns-init is in the init container #0. Append arguments at the end.
+     # pmem-driver is in the init container #0. Append arguments at the end.
```
Comment (pohly) [inline docs/install.md:344]: It's "the container #0", not "the *init* container #0".

File: pkg/pmem-device-manager/pmd-lvm.go [hunk 2]
```diff
@@ -287,3 +295,86 @@ func getVolumeGroups(groups []string) ([]vgInfo, error) {
 
 	return vgs, nil
 }
+
+func createNS(r *ndctl.Region, uselimit uint) error {
+	const align uint64 = 1024 * 1024 * 1024
+	realalign := align * r.InterleaveWays()
+	// uselimit is the percentage we can use
+	canUse := uint64(uselimit) * r.Size() / 100
```
Comment (pohly) [inline pkg/pmem-device-manager/pmd-lvm.go:303]: As you noticed here, `uselimit` is a name that cannot be understood without additional explanation. Why not just call the parameter "percentage"?

File: pkg/pmem-device-manager/pmd-lvm.go [hunk 3]
```diff
@@ -287,3 +295,86 @@ func getVolumeGroups(groups []string) ([]vgInfo, error) {
...
+	klog.V(3).Infof("Create fsdax-namespaces in %v, allowed %d %%, real align %d:\ntotal       : %16d\navail       : %16d\ncan use     : %16d",
+		r.DeviceName(), uselimit, realalign, r.Size(), r.AvailableSize(), canUse)
+	// Subtract sizes of existing active namespaces with currently handled mode and owned by pmem-csi
+	for _, ns := range r.ActiveNamespaces() {
+		klog.V(5).Infof("createNS: Exists: Size %16d Mode:%v Device:%v Name:%v", ns.Size(), ns.Mode(), ns.DeviceName(), ns.Name())
+		if ns.Name() == "pmem-csi" {
+			canUse -= ns.Size()
```
Comment (pohly) [inline pkg/pmem-device-manager/pmd-lvm.go:310]: Are you absolutely sure that the uint64 cannot underflow?

At some point in the past, the percentage may have been higher and a volume might have been created which is higher than the currently allowed maximum.

File: pkg/pmem-device-manager/pmd-lvm.go [hunk 4]
```diff
@@ -287,3 +295,86 @@ func getVolumeGroups(groups []string) ([]vgInfo, error) {
 
 	return vgs, nil
 }
+
+func createNS(r *ndctl.Region, uselimit uint) error {
```
Comment (pohly) [inline pkg/pmem-device-manager/pmd-lvm.go:299]: The logic in this function is complex enough that we should have a unit test which covers different scenarios.

Comment (avalluri) [reply pkg/pmem-device-manager/pmd-lvm.go:299]: The particular code was authored by @okartau. I just moved the existing code from `pmem-ns-init.go` to here.

Comment (pohly) [reply pkg/pmem-device-manager/pmd-lvm.go:299]: Then tests were already missing before. Now would still be a good time to add them.

Comment (avalluri) [reply pkg/pmem-device-manager/pmd-lvm.go:299]: Tests for this part of code is a bit complicated as the code depends on underlined PMEM hardware region properties(size, interleaves, max extent).  Current LVM unit tests are run on a pre-populated logical volume groups.

File: pkg/pmem-device-manager/pmd-lvm.go (hunk 3 above)
Comment (avalluri) [reply pkg/pmem-device-manager/pmd-lvm.go]: Fixed this case.

File: docs/install.md (hunk 1 above)
Comment (avalluri) [reply docs/install.md]: Fixed.

File: pkg/pmem-device-manager/pmd-lvm.go (hunk 2 above)
Comment (avalluri) [reply pkg/pmem-device-manager/pmd-lvm.go]: Renamed the variable.

File: pkg/pmem-device-manager/pmd-lvm.go (hunk 4 above)
Comment (pohly) [reply pkg/pmem-device-manager/pmd-lvm.go]: The usual approach for cases like this is to introduce an interface and then test the code with a mock implementation of the interface.

Comment (pohly) [reply pkg/pmem-device-manager/pmd-lvm.go]: Okay. Please file an issue to track it.

Comment (avalluri) [reply pkg/pmem-device-manager/pmd-lvm.go]: Filed #692

**Pull request merged**